src/runtime/meter-toast-api.js

```diff
@@ -65,18 +75,32 @@ export class MeterToastApi {
      */
     this.onConsumeCallback_ = null;
 
+    /**
+     * A timeout set while things are loading.  If the user dismisses the popup before
+     * loading is complete, this timeout ensures we consume the metering entitlement
+     * and cleanup the page properly.
+     * @private {?number} */
+    this.rapidCloseTimeout_ = null;
+
     /** @private @const {!function()} */
     this.sendCloseRequestFunction_ = () => {
+      if (this.rapidCloseTimeout_) {
+        this.win_.clearTimeout(this.rapidCloseTimeout_);
+        this.rapidCloseTimeout_ = null;
+        // TODO: Log some kind of 'timeout related to next logged event event'
+      } else {
+        const closeRequest = new ToastCloseRequest();
+        closeRequest.setClose(true);
+        this.activityIframeView_.execute(closeRequest);
+        this.removeCloseEventListener();
+      }
+
       this.deps_
         .eventManager()
         .logSwgEvent(
```

Comment:
By the TODO, it seems like you meant for this to go inside of the else. Is that right? This will still get logged on both timeout and regular close.

Reply:
I'd prefer to log something still.  You can tell it's the timeout still because you won't have a meter toast impression or offer event.

Reply:
LGTM